### PR TITLE
Added Azure Function and updated frontend to talk to it

### PR DESCRIPTION
**Pull request by neoKushan**
I've been tinkering around and I think I'm nearly there with the approach I wanted to go for, i.e. slimming it down to a straight-up function and having a dedicated frontend/SPA talk to it.

There's a lot going on with this PR, it's more a PoC than anything.

- I've created a "Core" library that just has some data models within it. This is just to separate concerns.
- I've added an Azure Function, using a cool library called "FunctionMonkey" to declaratively wire up a simple GET endpoint. It only returns dummy data but it'll be trivial to wire it up to the actual 15Five API. It also supports authentication (not added that yet).
-I've made a tweak to the frontend I created to call this Azure function API (Hardcoded for now, I just wanted to see it work) and it does work.

What would need to be done next is:

- Add auth to the function
- Integrate the 15Five handling as a separate service that the function can consume
- Make the frontend pretty af

### CODE REVIEW

File: Appreciation.AzureFunction/Function1.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Host;
+
+namespace Appreciation.Function
+{
+    public static class Function1
+    {
+        [FunctionName("Function1")]
+        public static async Task<HttpResponseMessage> Run([HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)]HttpRequestMessage req, TraceWriter log)
```
Comment (p0onage) [inline Appreciation.AzureFunction/Function1.cs]: These two functions are just default? can we remove these?

File: Appreciation.Function/Query/GetAppreciationsQuery.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using AppreciationApp.Core.Models;
+using AzureFromTheTrenches.Commanding.Abstractions;
+
+namespace AppreciationApp.Function.Query
+{
+    public class GetAppreciationsQuery : ICommand<List<Appreciation>>
+    {
+        public int DaysToLookBack { get; set; } = 7;
```
Comment (p0onage) [inline Appreciation.Function/Query/GetAppreciationsQuery.cs]: needs to be 6 days look back, 7 days will include the previous Friday ( we want to go Saturday to Friday)

Comment (neoKushan) [reply Appreciation.Function/Query/GetAppreciationsQuery.cs]: This is just a default value, you can pass in whatever you want to the api with ?daysToLookBack=<some value>. That way from the frontend you could do last week, last month, last year even (assuming the 15Five API goes back that far).

File: Appreciation.AzureFunction/Function1.cs (hunk 1 above)
Comment (neoKushan) [reply Appreciation.AzureFunction/Function1.cs]: Yeah this should not have gone in there! I thought I'd deleted it before pushing, my bad.

**Pull request merged**